packages/core/src/rules/oas3/no-illogical-composition-keywords.ts

```diff
@@ -0,0 +1,495 @@
+import { dequal } from '../../utils/dequal.js';
+import { isRef } from '../../ref-utils.js';
+import { isEmptyObject } from '../../utils/is-empty-object.js';
+import { isDefined } from '../../utils/is-defined.js';
+import { isPlainObject } from '../../utils/is-plain-object.js';
+
+import type { Oas3Rule, Oas3Visitor } from '../../visitors.js';
+import type { Oas3Schema, Oas3_1Schema } from '../../typings/openapi.js';
+import type { UserContext } from '../../walk.js';
+
+type SchemaSignature = {
+  properties?: Set<string>;
+  propertySchemas?: Map<string, SchemaSignature>;
+  enum?: string[];
+  items?: SchemaSignature;
+  additionalProperties?: boolean | SchemaSignature;
+  required?: Set<string>;
+  [key: string]: any;
+};
+
+type ReturnType = {
+  isExclusive: boolean;
+  reason?: string;
+};
+
+export const NoIllogicalCompositionKeywords: Oas3Rule = (): Oas3Visitor => {
+  return {
+    Schema: {
+      skip(node) {
+        return !node.oneOf && !node.anyOf && !node.allOf;
+      },
+      enter(schema, { report, location, resolve }) {
+        // Helper function to get the composition keyword and schemas
+        const compositionData = (() => {
+          if (schema.oneOf && Array.isArray(schema.oneOf)) {
+            return { keyword: 'oneOf' as const, schemas: schema.oneOf };
+          }
+          if (schema.anyOf && Array.isArray(schema.anyOf)) {
+            return { keyword: 'anyOf' as const, schemas: schema.anyOf };
+          }
+          if (schema.allOf && Array.isArray(schema.allOf)) {
+            return { keyword: 'allOf' as const, schemas: schema.allOf };
+          }
+          return null;
```

Comment:
It's literally undefined 🙂 

```suggestion
          return undefined;
```

Reply:
You are right